mirdata/datasets/cipi.py

```diff
@@ -0,0 +1,259 @@
+    and collaboration within the music education and research communities.
+"""
+import json
+import logging
+import os
+import pdb
+import pickle
+from typing import Optional, TextIO, List
+
+import smart_open
+from smart_open import open
+
+from deprecated.sphinx import deprecated
+
+from mirdata import core, io, jams_utils, download_utils
+
+try:
+    import music21
+except ImportError:
+    logging.error(
+        "In order to use cipi you must have music21 installed. "
+        "Please reinstall mirdata using `pip install 'mirdata[cipi]'"
+    )
+    raise ImportError
+
+BIBTEX = """
+@article{Ramoneda2024,
+  author    = {Pedro Ramoneda and Dasaem Jeong and Vsevolod Eremenko and Nazif Can Tamer and Marius Miron and Xavier Serra},
+  title     = {Combining Piano Performance Dimensions for Score Difficulty Classification},
+  journal   = {Expert Systems with Applications},
+  volume    = {238},
+  pages     = {121776},
+  year      = {2024},
+  doi       = {10.1016/j.eswa.2023.121776},
+  url       = {https://doi.org/10.1016/j.eswa.2023.121776}
+}"""
+
+INDEXES = {
+    "default": "1.0",
+    "test": "1.0",
+    "1.0": core.Index(filename="cipi_index_1.0.json"),
+}
+
+LICENSE_INFO = (
+    "Creative Commons Attribution Non Commercial Share Alike 4.0 International."
+)
+
+DOWNLOAD_INFO = """
+    Unfortunately the files of the CIPI dataset are available
+    for download upon request. After requesting the dataset, you will receive a
+    link to download the dataset. You must download scores.zip, embeddings.zip and index.json
+    copy the files into the folder:
+        > cipi/
+            > index.json
+            > embeddings.zip
+            > scores.zip
+    unzip embedding.zip and scores.zip and copy the CIPI folder to {}
+"""
+
+
+class Track(core.Track):
+    """Can I play it? (CIPI) track class
+
+    Args:
+        track_id (str): track id of the track
+
+    Attributes:
+        title (str): title of the track
+        book (str): book of the track
+        URI (str): URI of the track
+        composer (str): name of the author of the track
+        track_id (str): track id
+        musicxml_paths (list): path to musicxml score. If the music piece contains multiple movents the list will contain multiple paths.
+        difficulty annotation (str): annotated difficulty
+
+    Cached Properties:
+        Fingering path (str): Path of fingering features from technique dimension computed with ArGNN fingering model. Return of two paths, embeddings of the right hand and the ones of the left hand. Use torch.load(...) for loading the embeddings.
+        Expressiviness path (str): Path of expressiviness features from sound dimension computed with virtuosoNet model.Use torch.load(...) for loading the embeddings.
+        Notes path (str): Path of note features from notation dimension. Use torch.load(...) for loading the embeddings.
```

Comment:
Again, should these paths be in __init__? and then be Attributes instead of Cached Properties?